torchaudio/transforms.py

```diff
@@ -284,22 +291,102 @@ class SpectogramToDB(object):
     """
     def __init__(self, stype="power", top_db=None):
         self.stype = stype
-        if top_db is not None and top_db > 0:
-            top_db = -top_db
         self.top_db = top_db
         self.multiplier = 10. if stype == "power" else 20.
+        self.amin = torch.tensor([1e-10])
+        self.ref_value = torch.tensor([1.0])
 
     def __call__(self, spec):
 
-        spec_db = self.multiplier * torch.log10(spec / spec.max())  # power -> dB
+        # numerically stable implementation from librosa
+        # https://librosa.github.io/librosa/_modules/librosa/core/spectrum.html
+        spec_db = self.multiplier * torch.log10(torch.max(self.amin.to(spec.device), spec))
+        spec_db -= self.multiplier * torch.log10(torch.max(self.amin.to(spec.device), self.ref_value.to(spec.device)))
+
         if self.top_db is not None:
-            spec_db = torch.max(spec_db, spec_db.new_full((1,), self.top_db))
+            spec_db = torch.max(spec_db, spec_db.new_full((1,), spec_db.max() - self.top_db))
```

Comment:
The docstring needs a strong warning about using `top_db`, as it will make the output depend on the maximum value in the spectrogram -- e.g., you may get different outputs when feeding an audio file in snippets and concatenating it compared to feeding it as a whole. The same applies to `MFCC`, which uses a default `top_db` of 80.

Reply:
Would you suggest we make a relative vs. absolute parameter?